### PR TITLE
Make SonataEasyExtends optional

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is BC.

Part of: https://github.com/sonata-project/dev-kit/issues/750

SonataEasyExtends is deprecated but it is used on many bundles, this makes it optional. This change is BC because if the user does not change anything it will continue using the deprecated code but with a message.

## Changelog

```markdown
### Changed
- SonataEasyExtendsBundle is now optional, using SonataDoctrineBundle is preferred
### Deprecated
- Using SonataEasyExtendsBundle to add Doctrine mapping information
```

## To do

- [x] Update the documentation;
- [x] Add an upgrade note.

### CODE REVIEW

**Review by greg0ire — DISMISSED** [review]

**Comment by greg0ire** [comment]
Why the huge drop in coverage?

**Comment by jordisala1991** [comment]
the extension is not tested, I will try to add some test, those PR will be huge. I saw the recipes from flex and contain the Orm relations, so I will need to change them too, and create the missing ones.

**Comment by greg0ire** [comment]
:thinking: so is that PR finished or do you plan to do this later?

**Comment by jordisala1991** [comment]
It is not finished, I will change it to draft again. And it cannot be merged if the bug on extensions is not merged.

File: src/DependencyInjection/SonataArticleExtension.php
Comment (ChristianGripp) [inline src/DependencyInjection/SonataArticleExtension.php:48]: The dependency is required, but the check assumes it is an optional dependency. Maybe we should throw an Exception, if there is no `SonataDoctrineBundle` and no `SonataCoreBundle` registered.

**Review by core23 — DISMISSED** [review]

File: src/DependencyInjection/SonataArticleExtension.php
Comment (jordisala1991) [reply src/DependencyInjection/SonataArticleExtension.php:48]: I think the deprecation message should only be about EasyExtends and SonataDoctrineBundle, otherwise it could be confusing on how to proceed. You only need to take care of removing easy extends and enabling SonataDoctrineBundle if you want this deprecation to go away.

**Comment by jordisala1991** [comment]
> Why the huge drop in coverage?

The drop in coverage is due to not testing the deprecated easy extends code

**Comment by jordisala1991** [comment]
Blocked by: #173 too

**Review by VincentLanglet — APPROVED** [review]

File: UPGRADE-1.x.md
Comment (greg0ire) [inline UPGRADE-1.x.md:4]: We should try to remember this while releasing. Would be great to have a keyword and have the release tool warn us that we need to do something.

**Review by greg0ire — DISMISSED** [review]

File: src/DependencyInjection/SonataArticleExtension.php
Comment (greg0ire) [reply src/DependencyInjection/SonataArticleExtension.php:48]: @jordisala1991 please refrain from resolving conversations you don't open, it makes it harder for other contributors to spot such conversations and decide who they think is correct. Now I'm unsure if I should dismiss @core23 's review or not…

@core23 a dependency can be required, but in the case of a bundle it does not mean it will be registered, although it's unlikely.

> Maybe we should throw an Exception, if there is no SonataDoctrineBundle and no SonataCoreBundle registered.

Did you mean "or" instead of "and". BTW, isn't the core bundle deprecated and no longer supposed to be used?

> You only need to take care of removing easy extends and enabling SonataDoctrineBundle if you want this deprecation to go away.

What happens currently if the core bundle isn't registered?

Comment (jordisala1991) [reply src/DependencyInjection/SonataArticleExtension.php:48]: AFAIK there is no relation between the SonataDoctrineBundle and the CoreBundle. If you dont register CoreBundle you won't get any error related to this PR.

You could have errors if you don't register CoreBundle and don't register FormExtension / TwigExtension, but that's unrelated

Comment (jordisala1991) [reply src/DependencyInjection/SonataArticleExtension.php:48]: And I don't think we should throw an error on stable branch if the user is not registering any of: EasyExtendsBundle and SonataDoctrineBundle, because that could cause errors where before there were none. I left a comment to throw an error on master branch.

**Comment by greg0ire** [comment]
Thanks @jordisala1991 !

File: src/DependencyInjection/SonataArticleExtension.php
Comment (jordisala1991) [reply src/DependencyInjection/SonataArticleExtension.php:48]: Sorry for the resolving conversations thing @greg0ire , didnt want to cause troubles

Comment (greg0ire) [reply src/DependencyInjection/SonataArticleExtension.php:48]: Hey, don't worry mate! And thanks for the huge work BTW!

File: docs/reference/installation.rst
Comment (phansys) [inline docs/reference/installation.rst:22]: IMO, this section must be updated in order to show the remaining persistence bundles.

Comment (jordisala1991) [reply docs/reference/installation.rst:22]: I left only the implemented ones for the ArticleBundle

**Pull request merged**
